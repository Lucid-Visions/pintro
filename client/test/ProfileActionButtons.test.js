import React from "react";
import { create, act } from "react-test-renderer";
import {
  FollowMeButton,
  MessageMeButton,
  ExtrasButton,
  ViewConnectionsButton,
  NewPostButton,
} from "../src/components/ProfileActionButtons";


/* Test for the action buttons in the profile page
 * "FOLLOW ME", "MESSAGE ME", "NEW POST", "CONNECTIONS", "..."
 * onPress functions are mocked and being tested if called onPress of buttons
*/

describe("<FollowMeButton />", () => {
  let requestSent;
  const mockOnPress = jest.fn(() => (requestSent = true));

  test("snapshot of component with status PENDING", () => {
    const component = create(
      <FollowMeButton onPress={mockOnPress} text={"PENDING"} />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("snapshot of component with status NOT CONNECTED", () => {
    const component = create(
      <FollowMeButton onPress={mockOnPress} text={"NOT CONNECTED"} />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("test if onPress will send request", () => {
    const component = create(
      <FollowMeButton onPress={mockOnPress} text={"PENDING"} />
    );
    const instance = component.root;
    const button = instance.findByType("View");

    expect(requestSent).toBeFalsy();
    button.props.onClick();
    expect(requestSent).toBe(true);
  });
});

describe("<MessageMeButton />", () => {
  let openMessageRequestModal;
  const mockOnPress = jest.fn(() => (openMessageRequestModal = true));

  test("snapshot of component", () => {
    const component = create(
      <MessageMeButton onPress={mockOnPress} />
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("test if onPress will open message me modal", () => {
    const component = create(<MessageMeButton onPress={mockOnPress} />);
    const instance = component.root;
    const button = instance.findByType("View");

    expect(openMessageRequestModal).toBeFalsy();
    button.props.onClick();
    expect(openMessageRequestModal).toBe(true);
  });
});

describe("<ExtrasButton />", () => {
  let extrasClicked;
  const mockOnPress = jest.fn(() => (extrasClicked = true));

  test("snapshot of component with some action onPress", () => {
    const component = create(<ExtrasButton action={mockOnPress} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("test onPress with no function to change something", () => {
    const component = create(<ExtrasButton />);
    const instance = component.root;
    const button = instance.findByType("View");

    expect(extrasClicked).toBeFalsy();
    button.props.onClick();
    expect(extrasClicked).toBeFalsy();
  });

  test("test if onPress will work", () => {
    const component = create(<ExtrasButton action={mockOnPress} />);
    const instance = component.root;
    const button = instance.findByType("View");

    expect(extrasClicked).toBeFalsy();
    button.props.onClick();
    expect(extrasClicked).toBe(true);
  });
});

describe("<ViewConnectionsButton />", () => {
  let redirectedToConnections;
  const mockOnPress = jest.fn(() => (redirectedToConnections = true));

  test("snapshot of component", () => {
    const component = create(<ViewConnectionsButton onPress={mockOnPress} />).toJSON();
    expect(component).toMatchSnapshot();
  });

  test("test if onPress will redirect user to another page", () => {
    const component = create(<ViewConnectionsButton onPress={mockOnPress} />);
    const instance = component.root;
    const button = instance.findByType("View");

    expect(redirectedToConnections).toBeFalsy();
    button.props.onClick();
    expect(redirectedToConnections).toBe(true);
  });
});

describe("<NewPostButton />", () => {
    let newPostOpened;
    const mockOnPress = jest.fn(() => (newPostOpened = true));
  
    test("snapshot of component", () => {
      const component = create(<NewPostButton onPress={mockOnPress} />).toJSON();
      expect(component).toMatchSnapshot();
    });
  
    test("test if onPress will redirect user to write new post", () => {
      const component = create(<NewPostButton onPress={mockOnPress} />);
      const instance = component.root;
      const button = instance.findByType("View");
  
      expect(newPostOpened).toBeFalsy();
      button.props.onClick();
      expect(newPostOpened).toBe(true);
    });
  });
