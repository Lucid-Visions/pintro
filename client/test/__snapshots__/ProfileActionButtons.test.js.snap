// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExtrasButton /> snapshot of component with some action onPress 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "borderColor": "black",
      "borderRadius": 7.5,
      "borderWidth": 1,
      "height": 30,
      "opacity": 1,
      "padding": 3,
      "width": 40,
    }
  }
>
  <Text>
    ...
  </Text>
</View>
`;

exports[`<FollowMeButton /> snapshot of component with status NOT CONNECTED 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "marginRight": "5%",
      "opacity": 1,
      "width": "38.5%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 10,
        "marginLeft": "5%",
      }
    }
  >
    CONNECT
  </Text>
</View>
`;

exports[`<FollowMeButton /> snapshot of component with status PENDING 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "marginRight": "5%",
      "opacity": 1,
      "width": "38.5%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 10,
        "marginLeft": "5%",
      }
    }
  >
    PENDING
  </Text>
</View>
`;

exports[`<MessageMeButton /> snapshot of component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "marginRight": "5%",
      "opacity": 1,
      "width": "38.5%",
    }
  }
>
  <Text />
  <Text
    style={
      Object {
        "fontSize": 10,
        "marginLeft": "5%",
      }
    }
  >
    MESSAGE ME
  </Text>
</View>
`;

exports[`<NewPostButton /> snapshot of component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "black",
      "borderRadius": 5,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "marginRight": "5%",
      "opacity": 1,
      "width": "38.5%",
    }
  }
>
  <Text
    style={
      Object {
        "color": "white",
        "fontSize": 10,
        "marginLeft": "5%",
      }
    }
  >
    NEW POST
  </Text>
</View>
`;

exports[`<ViewConnectionsButton /> snapshot of component 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderColor": "black",
      "borderRadius": 5,
      "borderWidth": 1,
      "flexDirection": "row",
      "height": 30,
      "justifyContent": "center",
      "marginRight": "5%",
      "opacity": 1,
      "width": "38.5%",
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 10,
        "marginLeft": "5%",
      }
    }
  >
    CONNECTIONS
  </Text>
</View>
`;
